pandas/core/indexes/base.py

```diff
@@ -4281,7 +4288,9 @@ def get_value(self, series, key):
 
         # if we have something that is Index-like, then
         # use this, e.g. DatetimeIndex
-        s = getattr(series, '_values', None)
+        # changed from None to Series so that Series.at works.
+        # See if we can fix there.
+        s = getattr(series, '_values', series)
```

Comment:
Some context. With:

```python
In [5]: tzser = pd.Series(pd.date_range("2000", periods=2, tz="UTC"))

In [6]: tzser.at[0]
```

It seems that `Series._get_value` pass the EA directly

```python
1075        def _get_value(self, label, takeable=False):
1076            if takeable is True:
1077                return com.maybe_box_datetimelike(self._values[label])
1078 ->         return self.index.get_value(self._values, label)
```

So the type of the `series` argument to `Series.get_value` is at least `Union[Series, ExtensionArray]`.

I'm going to split this out.

Reply:
https://github.com/pandas-dev/pandas/pull/24204